frontend/src/features/search/state/searchState.ts

```diff
@@ -0,0 +1,34 @@
+import { createReducer } from '@reduxjs/toolkit'
+
+import { RootState } from '../../../store/store';
+import { SearchResult } from '../../../qri/searchResult'
+
+export const selectSearchResults = (state: RootState): SearchResult[] => state.search.searchResults
+
+export interface SearchState {
```

Comment:
add a `request: SearchParams` field to SearchState. We need to retain the `q` & `orderby` fields to know when to invalidate `searchResults` & `resultCount` by comparing the stored request with any new request.

To that end, add a `SET_SEARCH` action that we fire when search parameters change. Reducer should reset `searchResults` & `resultCount` within that action handler if params change